feat: add ADR redirects and SDK documentation restructuring

Cordtus opened the pull request:
- Add comprehensive ADR redirects to GitHub cosmos-sdk-docs repository
- Restructure SDK documentation for v0.53, v0.50, v0.47, and next versions
- Move tutorials to SDK next version
- Update navigation structure in docs.json
- Add migration and formatting scripts
- Clean up obsolete files and temporary artifacts

tac0turtle commented:
can you run vangen to generate the updated files

julienrbrt reviewed (commented):
Can you delete this please: https://github.com/cosmos/vanity/tree/main/tests

Cordtus committed:
feat: add ADR redirects and SDK documentation restructuring

- Add comprehensive ADR redirects to GitHub cosmos-sdk-docs repository
- Restructure SDK documentation for v0.53, v0.50, v0.47, and next versions
- Move tutorials to SDK next version
- Update navigation structure in docs.json
- Add migration and formatting scripts
- Clean up obsolete files and temporary artifacts

Cordtus committed:
fix breaking stuff
